src/components/popover-container/popover-container.pw.tsx

```diff
@@ -326,130 +323,137 @@ test.describe("Check props of Popover Container component", () => {
+    await openButton.click();
+    const popoverContainer = page.getByRole("dialog", {
+      name: "String is awesome",
+    });
+    await popoverContainer.press("Escape");
 
-    await expect(popoverContainerContentElement).not.toBeVisible();
+    await expect(popoverContainer).toBeHidden();
   });
 
-  test("should not close when an option is selected from Select component inside", async ({
+  test("should not close when an option is selected from a Select component inside", async ({
     mount,
     page,
   }) => {
     await mount(<PopoverContainerWithSelect />);
 
-    await popoverSettingsIcon(page).click();
-    await selectText(page).click();
-    await selectListText(page, "green").click();
+    const openButton = page.getByRole("button", { name: "open" });
+    await openButton.click();
+    const popoverContainer = page.getByRole("dialog", {
+      name: "select example",
+    });
+    const select = page.getByText("Please Select...", { exact: true });
+    await select.click();
+    const greenOption = page.getByRole("option", { name: "green" });
+    await greenOption.click();
 
-    await expect(popoverContainerContent(page)).toBeVisible();
+    await expect(popoverContainer).toBeVisible();
   });
 
-  // Due to a bug in Playwright whereby the Escape key fires a keyup event before the keydown event.
-  // This test is currently failing because the Select List closes on an Esc keyup event but the Popover-Container listens for a keydown event.
-  // The fix for this bug has been released in Playwright v1.40.0.
-  // FE-6349
-  // https://github.com/microsoft/playwright/pull/27711
-  test.skip("should not close when the escape key is pressed and the Select List is open", async ({
+  test("should not close when the escape key is pressed and the Select List is open", async ({
     mount,
     page,
   }) => {
     await mount(<PopoverContainerWithSelect />);
 
-    await popoverSettingsIcon(page).click();
-    const selectTextElement = selectText(page);
-    await selectTextElement.click();
-    const selectListElement = selectListText(page, "red");
-    await waitForAnimationEnd(selectListElement);
-    await page.keyboard.press("ArrowDown");
-    await page.keyboard.press("ArrowDown");
-    await popoverContainerComponent(page).press("Escape");
-
-    await expect(selectListText(page, "green")).not.toBeVisible();
-    await expect(popoverContainerContent(page)).toBeVisible();
+    const openButton = page.getByRole("button", { name: "open" });
+    await openButton.click();
+    const popoverContainer = page.getByRole("dialog", {
+      name: "select example",
+    });
+    const select = page.getByText("Please Select...", { exact: true });
+    await select.click();
+    await select.press("Escape");
+
+    await expect(popoverContainer).toBeVisible();
   });
 
-  test("should close when the escape key is pressed with focus on the Select component", async ({
```

Comment:
Removed this test since we have previously covered close behaviour on Escape key press

Reply:
Agree with Sian, this is to catch a regression where the Select previously didn't allow the event to bubble when it was closed.